static/alert.js

```diff
@@ -37,11 +37,14 @@ define(function (require) {
         }, this));
     }
 
-    Alert.prototype.alert = function (title, body) {
+    Alert.prototype.alert = function (title, body, onClose) {
         var modal = $('#alert');
         modal.find('.modal-title').html(title);
         modal.find('.modal-body').html(body);
         modal.modal();
+        if (onClose) {
```

Comment:
We can't do this, unfortunately. There is exactly one '#alert' box and this means every time someone makes an alert we chain another onClose() call on to it! It's not obvious I realise!

We'll need to have one 'onClose' handler defined in `Alert` that then checks a member variable `onClose` (like `yesHandler` and `noHandler`) and calls it if non-null. We can do this post-merge.